runtime/float.go

```diff
@@ -259,8 +261,14 @@ func floatRDivMod(f *Frame, v, w *Object) (*Object, *BaseException) {
 	})
 }
 
+const floatReprPrecision = 16
+
 func floatRepr(f *Frame, o *Object) (*Object, *BaseException) {
-	return NewStr(strconv.FormatFloat(toFloatUnsafe(o).Value(), 'g', -1, 64)).ToObject(), nil
+	v := unsignPositiveInf(strings.ToLower(strconv.FormatFloat(toFloatUnsafe(o).Value(), 'g', floatReprPrecision, 64)))
+	if noDecimal, _ := regexp.MatchString(`^\d+$`, v); noDecimal {
```

Comment:
This check seems pretty heavyweight for what we need to do here. At the very least we should compile the expression so that we don't do that in every call. But I think avoiding the regexp dependency altogether by using strings.IndexFunc() or something would be cleaner.